### PR TITLE
Introduce styledOmitProps, remove clean-tag and clean-element

### PR DESCRIPTION
**Pull request by philipwindeyer**
Removes clean-tag and clean-element.
Removes use of blacklist, and introduces styledOmitProps (which accepts a list of props to `omit` from a components rendered DOM node).

So styled-components does indeed filter out non-valid props on standard components.
However if we’re using forwardRef and createElement with a list of filtered props (i.e. our list of props to omit out), the validity check is not run on new components, hence styled-system props etc will make it onto the final DOM node - in this scenario specifically.
I haven’t made the time for fully understanding why that condition is not met in StyledComponents when we create a component this way (i.e. see: https://github.com/styled-components/styled-components/blob/e6f1b85726512cde163dd5fcdd7fb0f2439d4a09/packages/styled-components/src/models/StyledComponent.js#L152) , but the behaviour I’m seeing means that condition isn’t met.

To get around this, I'm using is-valid-prop from emotion in our implementation (which is exactly what happens in styled-components):

```js
  const omittedPropsComponent = React.forwardRef((props, ref) => {
    const validProps = pickBy(props, (_, prop) => isPropValid(prop));
    const filteredProps = omit(validProps, options.omit);
    return React.createElement(component, { ...filteredProps, ref });
  });
```
For reference, the list of validProps is here: https://github.com/emotion-js/emotion/blob/master/packages/is-prop-valid/src/props.js

Lots of back and forth on this, and frequent discussion with @talbet to get to this solution.
Would love some feedback, thoughts and testing. It's tested AF, but it could use a few more pairs of eyes given the size.
TIA ❤️

### CODE REVIEW

File: packages/components/src/DatePicker/components/CalendarMonth/CalendarMonth.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-import styled, { css } from 'styled-components';
-
+import { css } from 'styled-components';
 import { Box, Text } from '../../../';
 import CalendarWeekdays from '../CalendarWeekdays';
 import CalendarDays from '../CalendarDays';
+import { styledOmitProps } from '../../../styledOmitProps';
 
-const MonthWrapper = styled(Box)`
+const MonthWrapper = styledOmitProps(Box, { omit: ['monthsToDisplay', 'stacked'] })`
```
Comment (angusfretwell) [inline packages/components/src/DatePicker/components/CalendarMonth/CalendarMonth.js]: `styled(({ monthsToDisplay, stacked, ...rest }) => <Box {...rest} />)`

File: packages/components/src/OutlineButton/OutlineButton.test.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ describe('<OutlineButton />', () => {
     );
   });
 
-  fit('renders correctly', () => {
```
Comment (talbet) [inline packages/components/src/OutlineButton/OutlineButton.test.js]: Good catch, maybe this should be linted in the future

Comment (angusfretwell) [reply packages/components/src/OutlineButton/OutlineButton.test.js]: totes

**Comment by angusfretwell** [comment]
@talbet I think its OK for now

File: packages/components/src/OutlineButton/OutlineButton.test.js (hunk 2 above)
Comment (philipwindeyer) [reply packages/components/src/OutlineButton/OutlineButton.test.js]: Will throw a card up

**Comment by philipwindeyer** [comment]
Thank you @angusfretwell @talbet !
I'll merge now. There's cards up for the next steps (i.e. what was discussed in WG yesterday)

**Pull request merged**